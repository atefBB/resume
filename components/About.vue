<template>
  <div v-if="about.about_content" class="container">
    <h3 id="about-title">
      {{ about.about_title ? about.about_title : "About Me" }}
    </h3>
    <el-row v-if="about.about_profile_image">
      <el-col :span="8">
        <p>
          <el-avatar
            id="about-profile-image"
            shape="circle"
            fit="fill"
            :src="about.about_profile_image"
          />
        </p>
      </el-col>
      <el-col :span="16">
        <!-- eslint-disable-next-line -->
        <p id="about-content" v-html="about.about_content" />
      </el-col>
    </el-row>
    <el-row v-else>
      <el-col :span="24">
        <!-- eslint-disable-next-line -->
        <p id="about-content" v-html="about.about_content" />
      </el-col>
    </el-row>
  </div>
</template>

<script>
import about from "~/assets/yaml/about.yml"
import { safeLoad } from "js-yaml"

export default {
  props: [],
  data() {
    return {
      about: null
    }
  },
  created: function() {
    this.getContent()
  },
  methods: {
    getContent() {
      this.about = safeLoad(about)
    }
  }
}
</script>
