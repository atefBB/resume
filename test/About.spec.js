import faker from "faker"
import { testHelper, mountHelper } from "./helpers"
import About from "~/components/About.vue"
jest.mock(`~/assets/yaml/about.yml`)

describe("About", () => {
  testHelper(About)

  describe("Matches randoms values", () => {
    const values = {
      about: {
        about_title: faker.hacker.adjective(),
        about_content: faker.lorem.paragraph(),
        about_profile_image: faker.image.avatar()
      }
    }
    const mockFn = jest.fn()
    const options = {
      methods: {
        getContent: mockFn
      }
    }

    const factory = mountHelper(About)

    it("Should call the mocked function", () => {
      factory({ options, values })
      expect(mockFn).toHaveBeenCalled() //we use correctly the mock function
    })

    //about_title
    it("Should contain the about_title", () => {
      const wrapper = factory({ options, values })
      expect(wrapper.find("#about-title").text()).toEqual(
        expect.stringContaining(values.about.about_title)
      )
    })
    it("Should contain the default about_title", () => {
      let localValues = { about: { ...values.about, about_title: undefined } }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("#about-title").text()).toEqual(
        expect.stringContaining("About Me")
      )
    })

    //about_content
    it("Should contain the about_content", () => {
      const wrapper = factory({ options, values })
      expect(wrapper.find("#about-content").text()).toEqual(
        expect.stringContaining(values.about.about_content)
      )
    })
    it("Should NOT contain the about section", () => {
      let localValues = { about: { ...values.about, about_content: undefined } }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.isEmpty()).toBe(true)
    })

    //about_profile_image
    it("Should contain the about_profile_image", () => {
      const wrapper = factory({ options, values })
      expect(
        wrapper.find("#about-profile-image img").attributes("src")
      ).toEqual(expect.stringContaining(values.about.about_profile_image))
    })
    it("Should NOT contain the about_profile_image", () => {
      let localValues = {
        about: { ...values.about, about_profile_image: undefined }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("#about-profile-image").exists()).toBe(false)
    })
  })
})
