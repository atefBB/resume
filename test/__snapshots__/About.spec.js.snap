// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Default Template 1`] = `
<div class="container">
  <h3 id="about-title">
    About Me
  </h3>
  <div class="el-row">
    <div class="el-col el-col-8">
      <p><span class="el-avatar el-avatar--circle" id="about-profile-image"><img src="me.jpg" style="object-fit: fill;"></span></p>
    </div>
    <div class="el-col el-col-16">
      <p id="about-content">Hi, my nameâ€™s David. I'm Web Developer. My experience has allowed me to acquire solid knowledge in development, but not only. Whatever programming language is used, my knowledge of software engineering allows me to put into practice, with rigor, the best development practices. Curious by nature and self-taught I like to follow the new technologies news. Continuous learning is an indispensable thing for me, both professionally and personally.
      </p>
    </div>
  </div>
</div>
`;
